Kuroba/app/src/main/java/com/github/adamantcheese/chan/report/ReportProblemController.kt

```diff
@@ -0,0 +1,154 @@
+    private lateinit var reportActivityLogsText: TextInputEditText
+    private lateinit var reportActivityCancel: AppCompatButton
+    private lateinit var reportActivitySendReport: AppCompatButton
+
+    private val presenter: ReportProblemPresenter = ReportProblemPresenter()
+    private var loadingViewController: LoadingViewController? = null
+    private var onFinishedListener: (() -> Unit)? = null
+
+    override fun onCreate() {
+        Chan.inject(this)
+
+        view = inflate(context, R.layout.activity_report).apply {
+            reportActivityProblemTitle = findViewById(R.id.report_activity_problem_title)
+            reportActivityProblemDescription = findViewById(R.id.report_activity_problem_description)
+            reportActivityAttachLogsButton = findViewById(R.id.report_activity_attach_logs_button)
+            reportActivityLogsText = findViewById(R.id.report_activity_logs_text)
+            reportActivityCancel = findViewById(R.id.report_activity_cancel)
+            reportActivitySendReport = findViewById(R.id.report_activity_send_report)
+        }
+
+        presenter.onCreate(this)
+        compositeDisposable = CompositeDisposable()
+
+        val logs = LogsController.loadLogs()
+        if (logs != null) {
+            reportActivityLogsText.setText(logs)
+        }
+
+        reportActivityAttachLogsButton.setOnCheckedChangeListener { _, isChecked ->
+            reportActivityLogsText.isEnabled = isChecked
+        }
+        reportActivityCancel.setOnClickListener { onFinishedListener?.invoke() }
+        reportActivitySendReport.setOnClickListener { onSendReportClick() }
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+
+        onFinishedListener = null
+        presenter.onDestroy()
+        compositeDisposable.dispose()
+    }
+
+    fun setOnFinishedCallback(callback: () -> Unit) {
+        this.onFinishedListener = callback
+    }
+
+    private fun onSendReportClick() {
+        val title = reportActivityProblemTitle.text?.toString() ?: ""
+        val description = reportActivityProblemDescription.text?.toString() ?: ""
+        val logs = reportActivityLogsText.text?.toString() ?: ""
+
+        if (title.isEmpty()) {
+            reportActivityProblemTitle.error =
+                    context.getString(R.string.report_activity_title_cannot_be_empty_error)
+            return
+        }
+
+        if (description.isEmpty()) {
+            reportActivityProblemDescription.error =
+                    context.getString(R.string.report_activity_description_cannot_be_empty_error)
+            return
+        }
+
+        if (reportActivityAttachLogsButton.isChecked && logs.isEmpty()) {
+            reportActivityLogsText.error =
+                    context.getString(R.string.report_activity_logs_are_empty_error)
+            return
+        }
+
+        val logsParam = if (!reportActivityAttachLogsButton.isChecked) {
```

Comment:
Use a ternary operator here, it's cleaner.

Reply:
There is no ternary operator in kotlin, haha.

Reply:
And you don't need it since if/else expression result can be assigned to a variable.

Reply:
Gross.

Reply:
How is it gross? It's infinitelly better that doing something like:

String test = null;
if (x) {
  test = "123";
else {
  test = "456";
}

And you can't even mark "test" as "final" so it will stay mutable which is really bad.

And that's not all. You can do the same with the when expression (improved switch), try/catch/finally etc. You can't do it only with loops. Everything else is assignable.